### PR TITLE
Refactoring pg_dump extract schema and skipping unnecessary GRANT/REVOKE sql statements

### PR DESCRIPTION
**Pull request by sanyamsinghal**
Changes in this patch:
1. Code refactoring for pg_dump extract schema to a clean form
2. Adding `--no-privileges` flag to `pg_dump` to avoid exporting user privileges on the exported database objects.

### CODE REVIEW

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go [hunk 1]
```diff
@@ -55,57 +55,59 @@ func parseSchemaFile(exportDir string) {
...
-				createMatViewSqls.WriteString(sqlStatement)
-			case "SCHEMA":
-				createSchemaSqls.WriteString(sqlStatement)
+				objSqlStmts["MVIEW"].WriteString(sqlStatement)
 			case "EXTENSION":
-				createExtensionSqls.WriteString(sqlStatement)
+				objSqlStmts["EXTENSION"].WriteString(sqlStatement)
 			default:
 				uncategorizedSqls.WriteString(sqlStatement)
 			}
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/pg_dump_extract_schema.go:113]: switch sqlType {
			case "SCHEMA", "TYPE", "DOMAIN", "SEQUENCE", "INDEX", "RULE", "FUNCTION", "AGGREGATE", "PROCEDURE", "VIEW", "TRIGGER", "EXTENSION":
				objSqlStmts[sqlType].WriteString(sqlStatement)

			case "TABLE", "DEFAULT", "CONSTRAINT", "FK CONSTRAINT":
				objSqlStmts["TABLE"].WriteString(sqlStatement)

			case "MATERIALIZED VIEW":
				objSqlStmts["MVIEW"].WriteString(sqlStatement)

			default:
				uncategorizedSqls.WriteString(sqlStatement)
			}

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go [hunk 2]
```diff
@@ -120,31 +122,21 @@ func parseSchemaFile(exportDir string) {
 		}
 	}
 
-	//TODO: convert below code into a for-loop
+	for objType, sqlStmts := range objSqlStmts {
+		if sqlStmts.Len() > 0 { // create .sql file only if there are DDLs
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/pg_dump_extract_schema.go:126]: Invert the condition to reduce the indentation level of the code:

```
for .... {
    if sqlStmts.Len() == 0 {
        continue
    }
    filePath := ...
    // Continue with further code.
}
```

**Review by amit-yb — APPROVED** [review]
Just a couple of minot nits.

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 1 above)
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:113]: good catch

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 2 above)
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go]: done

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 1 above)
Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go]: done

**Pull request merged**